tests/tenant_test.go

```diff
@@ -0,0 +1,262 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/kr/pretty"
+)
+
+var (
+	stagingInstancesNumber    = 2
+	prodInstancesNumber       = 6
+	stagingInstanceNamePrefix = "testInstanceStaging"
+	prodInstanceNamePrefix    = "testInstanceProd"
+)
+
+func TestTenantSchemaUpdate(t *testing.T) {
+	ctx := context.Background()
+	ctl := &controller{}
+	dataDir := t.TempDir()
+	if err := ctl.StartMain(ctx, dataDir); err != nil {
+		t.Fatal(err)
+	}
+	defer ctl.Close()
+	if err := ctl.Login(); err != nil {
+		t.Fatal(err)
+	}
+	if err := ctl.setLicense(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a project.
+	project, err := ctl.createProject(api.ProjectCreate{
+		Name:       "Test Project",
+		Key:        "TestSchemaUpdate",
+		TenantMode: api.TenantModeTenant,
+	})
+	if err != nil {
+		t.Fatalf("failed to create project, error: %v", err)
+	}
+
+	// Provision an instance.
+	instanceRootDir := t.TempDir()
+
+	var stagingInstanceDirs []string
+	var prodInstanceDirs []string
+	for i := 0; i < stagingInstancesNumber; i++ {
+		instanceDir, err := ctl.provisionSQLiteInstance(instanceRootDir, fmt.Sprintf("%s-%d", stagingInstanceNamePrefix, i))
+		if err != nil {
+			t.Fatal(err)
+		}
+		stagingInstanceDirs = append(stagingInstanceDirs, instanceDir)
+	}
+	for i := 0; i < prodInstancesNumber; i++ {
+		instanceDir, err := ctl.provisionSQLiteInstance(instanceRootDir, fmt.Sprintf("%s-%d", prodInstanceNamePrefix, i))
+		if err != nil {
+			t.Fatal(err)
+		}
+		prodInstanceDirs = append(prodInstanceDirs, instanceDir)
+	}
+	environments, err := ctl.getEnvironments()
+	if err != nil {
+		t.Fatal(err)
+	}
+	stagingEnvironment, err := findEnvironment(environments, "Staging")
+	if err != nil {
+		t.Fatal(err)
+	}
+	prodEnvironment, err := findEnvironment(environments, "Prod")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add an instance.
+	var stagingInstances []*api.Instance
+	var prodInstances []*api.Instance
+	for i, stagingInstanceDir := range stagingInstanceDirs {
+		instance, err := ctl.addInstance(api.InstanceCreate{
+			EnvironmentID: stagingEnvironment.ID,
+			Name:          fmt.Sprintf("%s-%d", stagingInstanceNamePrefix, i),
+			Engine:        db.SQLite,
+			Host:          stagingInstanceDir,
+		})
+		if err != nil {
+			t.Fatalf("failed to add instance, error: %v", err)
+		}
+		stagingInstances = append(stagingInstances, instance)
+	}
+	for i, prodInstanceDir := range prodInstanceDirs {
+		instance, err := ctl.addInstance(api.InstanceCreate{
+			EnvironmentID: prodEnvironment.ID,
+			Name:          fmt.Sprintf("%s-%d", prodInstanceNamePrefix, i),
+			Engine:        db.SQLite,
+			Host:          prodInstanceDir,
+		})
+		if err != nil {
+			t.Fatalf("failed to add instance, error: %v", err)
+		}
+		prodInstances = append(prodInstances, instance)
+	}
+
+	// Set up label values for tenants. prodInstancesNumber is the number of tenants.
```

Comment:
any particular reason we skip staging instance?

Reply:
We're using the same tenant names. Updated the comment.